netmaster/master/network.go

```diff
@@ -338,18 +363,20 @@ func DeleteNetworkID(stateDriver core.StateDriver, netID string) error {
 		return err
 	}
 
-	// Check if there are any active endpoints
-	if hasActiveEndpoints(nwCfg) {
-		return core.Errorf("Error: Network has active endpoints")
-	}
-
-	if GetClusterMode() == "docker" {
+	if IsDNSEnabled() {
 		// detach Dns container
 		err = detachServiceContainer(nwCfg.Tenant, nwCfg.NetworkName)
 		if err != nil {
 			log.Errorf("Error detaching service container. Err: %v", err)
 		}
+	}
 
+	// Check if there are any active endpoints
+	if hasActiveEndpoints(nwCfg) {
```

Comment:
/cc @vvb: With the DNS option being explicit now, we should ideally be detaching the container from the network. After this we should check if there are no active endpoints whether dns is enabled or not. Made the corresponding changes in hasActiveEndpoints function as well. Please check it out and let me know if you there are any scenarios that might break your changes.